src/gui/qgsadvanceddigitizingcanvasitem.cpp

```diff
@@ -59,6 +59,13 @@ void QgsAdvancedDigitizingCanvasItem::paint( QPainter *painter )
     }
   }
 
+  // Draw current intersection tool
+  if ( mAdvancedDigitizingDockWidget->tool() )
```

Comment:
```suggestion
  if ( QgsAdvancedDigitizingTool * tool = mAdvancedDigitizingDockWidget->tool() )
  {
      // if a tool is active in the dock, then delegate to that tool to handle decorating the canvas instead of using the default decorations
      tool->paint( painter );
      return;
  }
```

(Maybe in future we could add some flag for the tool "draw default CAD decorations", when there's a need to)